test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,212 @@
+package firewall
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+
+	"github.com/openshift-kni/commatrix/pkg/types"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+func ApplyRulesToNode(cs *client.ClientSet, ns string, NFTtable []byte, nodeName string, wg *sync.WaitGroup) error {
+	defer wg.Done()
+
+	utilsHelpers := utils.New(cs)
+
+	debugPod, err := utilsHelpers.CreatePodOnNode(nodeName, ns, consts.DefaultDebugPodImage)
+	if err != nil {
+		return fmt.Errorf("failed to create debug pod on node %s: %w", nodeName, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	command := []string{
+		"bash", "-c",
+		fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+	}
+
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to save rule set to file on node %s: %w", nodeName, err)
+	}
+
+	if err := installNFT(debugPod, utilsHelpers); err != nil {
+		return err
+	}
+
+	command = []string{
+		"bash", "-c", "/usr/sbin/nft -f nft/firewall.nft",
+	}
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to apply rule set %q on node %s: %w", command, nodeName, err)
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	command = []string{
+		"bash", "-c",
+		"cp nft/firewall.nft /host/etc/nftables/firewall.nft",
+	}
+
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to move NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	err = editNftablesConf(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	command = []string{
+		"bash", "-c",
+		"/usr/sbin/nft list ruleset > /etc/nftables.conf",
+	}
+
+	_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+	if err != nil {
+		return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	command = []string{"bash", "-c",
+		"systemctl enable nftables",
```

Comment:
please document this one that is not supported in openshift.

you can't just enable a system service this will not survive a cluster upgrade when applying a new RHCOS image we need to find a different solution I think.

or understand how we expect the users to configure the nftables on the machines

Reply:
cc @yuvalk

Reply:
its from here 
https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/configuring_firewalls_and_packet_filters/getting-started-with-nftables_firewall-packet-filters#including-files-in-nftables-scripts_writing-and-executing-nftables-scripts